Unexpand: Initial implementation

bojan88 opened the pull request:
Implementation for https://github.com/GrayJack/coreutils/issues/7

GrayJack commented on unexpand/src/tab_stops.rs:
```diff
@@ -0,0 +1,183 @@
+                    tabs_str.split(|c| c == ',' || c == ' ').map(|s| s.trim()).collect();
+
+                if tabs_vec.len() == 1 {
+                    let value = tabs_vec[0].parse::<usize>().unwrap();
```
There is a awful lot of unwrapping in the code, there is any guarantee that these uses of unwrap won't panic?

GrayJack commented on unexpand/src/unexpand.yml:
```diff
@@ -0,0 +1,24 @@
+about: Convert spaces to tabs
+args:
+    - all:
+        help: convert all blanks, instead of just initial blanks
```
All help start with upper case

GrayJack commented on unexpand/src/unexpand.yml:
```diff
@@ -0,0 +1,24 @@
+        takes_value: true
+    - version:
+        help: output version information and exit
+        long: version
```
I think that is not needed (help and version are added by default on clap)

GrayJack commented on unexpand/src/unexpand.yml:
```diff
@@ -0,0 +1,24 @@
+        help: convert all blanks, instead of just initial blanks
+        short: a
+        long: all
+    - first_only:
```
All options/flags should have a short and a long one, it it doesn't have on the original, you are free to create one

bojan88 commented:
@GrayJack thanks for the quick review. I appreciate it, especially the error handling parts.
I hope I addressed all your suggestions/requests.
Can you please review it again?

bojan88 commented on unexpand/src/tab_stops.rs:
```diff
@@ -0,0 +1,210 @@
+const ARG_PARSE_MSG: &str = "unexpand: error parsing arguments";
+
+impl TabStops {
+    pub fn new(tabs_str: Option<&str>) -> Result<TabStops, &str> {
```
I'm not sure, but I have a feeling that `new` should always return the structure instance.
Maybe I should rename this method...

GrayJack replied:
Nah, that is ok in cases where the struct cannot be safely constructed  without check, as example, checkout `CString::new()` on `std`

GrayJack commented:
Looks fine for me :smile: 

The test grow big, so I would recommend create a module just for tests, give a look at the `wc` and `cut`, that's the way I did it the tests module

bojan88 commented:
@GrayJack I moved the tests to a separate module, but in the same file. Hope that's fine...

GrayJack commented:
It's great!! Thanks for the contribution
